Add release notes for OData Connected Service version 0.12.1 and other skipped versions (#173)
MicrosoftDocs/OData-docs: Odata-docs/changelog/odataconnectedservice.md

# OData Connected Service changelog

OData Connected Service extension is available on [Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=marketplace.ODataConnectedService).

## OData Connected Service 0.12.1

### Improvements and Bug Fixes:
* [ [#202](https://github.com/OData/ODataConnectedService/pull/202) ] Fix invalid escaping of double quotes when generating VB.NET client proxy

> * [ [#203](https://github.com/OData/ODataConnectedService/pull/203) ] Fix non-compilable code emitted when the option to generate multiple files is checked and the model contains 2 types with the same name but in different namespace

Reviewer: ```suggestion
* [ [#203](https://github.com/OData/ODataConnectedService/pull/203) ] Fix uncompilable code emitted when the option to generate multiple files is checked and the model contains 2 types with the same name but in different namespace
```

* [ [#209](https://github.com/OData/ODataConnectedService/pull/209) ] Restore namespace in the metadata document as the default namespace applied in the client proxy

> * [ [#203](https://github.com/OData/ODataConnectedService/pull/203) ] Fix non-compilable code emitted when the option to generate multiple files is checked and the model contains 2 types with the same name but in different namespace

Reviewer 2: Should these be in any particular order? (alphabetical, pullId, date of merge, etc?)
If not lgtm, if yes, probably applies for all the other entries...

Author: @g2mula Traditionally we have never bothered about any order. Alphabetic would be a bit hard to manage since descriptions tend to mutate often. Date of merge would be tedious to keep track of. We could maybe adopt pull request ID to create some order, and since it doesn't hurt I'll make that change

## OData Connected Service 0.12.0

### Improvements and Bug Fixes:
* [ [#178](https://github.com/OData/ODataConnectedService/pull/178) ] Fix regression errors when generating code for V1-V3 services

> * [ [#178](https://github.com/OData/ODataConnectedService/pull/178) ] Fix regression errors caused by UI updates not running in UI thread

Reviewer 3: Seems like 178 is repeate multiple times maybe merge the descripions?

* [ [#133](https://github.com/OData/ODataConnectedService/pull/133) ] Fix issue where `false` is generated as `False` (uppercase F) in C# code, causing errors
* [ [#135](https://github.com/OData/ODataConnectedService/pull/135) ] Include enum types from enum collection properties in generated code to avoid compilation errors
* [ [#148](https://github.com/OData/ODataConnectedService/pull/148) ] Rename generated parameters source and keys to avoid name collisions with similar names in the model
* [ [#154](https://github.com/OData/ODataConnectedService/pull/154) ] Fix issue causing undefined Edmx reference to appear in generated code
* [ [#160](https://github.com/OData/ODataConnectedService/pull/160) ] Fix build error caused by incompatible `System.Text.Json` version with new versions of OData client

## OData Connected Service 0.10.0

### New Features:
* [ [#83](https://github.com/OData/ODataConnectedService/pull/83) ] Emit dynamic properties container property for open types

### Improvements and Bug Fixes:
* [ [#116](https://github.com/OData/ODataConnectedService/pull/116) ] Properly escapes metadata path in the local filesystem when generating T4 templates

> * [ [#117](https://github.com/OData/ODataConnectedService/pull/117) ] Fix issue where unselecting an operation import auto-selects schema types that the operation depends o

Reviewer: ```suggestion
* [ [#117](https://github.com/OData/ODataConnectedService/pull/117) ] Fix issue where unselecting an operation import auto-selects schema types that the operation depends on
```

> * [ [#121](https://github.com/OData/ODataConnectedService/pull/121) ] Fix issue loading generated metadata file when you have multiple files generated in the client projectn

Reviewer: ```suggestion
* [ [#121](https://github.com/OData/ODataConnectedService/pull/121) ] Fix issue loading generated metadata file when you have multiple files generated in the client project
```

* [ [#168](https://github.com/OData/ODataConnectedService/pull/168) ] Generate doc comments in generated code from description annotations in the OData schema
* [ [#145](https://github.com/OData/ODataConnectedService/pull/145) ] Display the number of selected operation imports

### Improvements and Bug Fixes:
* [ [#148](https://github.com/OData/ODataConnectedService/pull/148) ] Rename generated parameters source and keys to avoid name collisions with similar names in the model
* [ [#160](https://github.com/OData/ODataConnectedService/pull/160) ] Fix build error caused by incompatible `System.Text.Json` version with new versions of OData client
* [ [#135](https://github.com/OData/ODataConnectedService/pull/135) ] Include enum types from enum collection properties in generated code to avoid compilation errors
* [ [#133](https://github.com/OData/ODataConnectedService/pull/133) ] Fix issue where `false` is generated as `False` (uppercase F) in C# code, causing errors
* [ [#154](https://github.com/OData/ODataConnectedService/pull/154) ] Fix issue causing undefined Edmx reference to appear in generated code

> ## OData Connected Service 0.3.1 (Legacy)

Reviewer: Should this be between 0.10 and 0.11?

Author: @wandeg @marabooy The thing is, that release was created by @habbes in between 0.10.0 and 0.11.0. It was not targeted at the active version at the time. I could move it so its appear in the right order version-wise but that order would be misleading chronologically. It appears between 0.10.0 and 0.11.0 in GitHub releases so I'd curious to get an opinion from @habbes (and any other person who might have one) before making the change...

Reviewer 4: For the release notes on MS docs, we don't need the `0.3.1 legacy` notes

Reviewer 5: @gathogojr I think the chronological listing makes sense. But if it's too confusing, then I don't think there's an issue placing it at the bottom. I guess someone looking for a specific version would search the changelog in the order of versions, not chronologically (since they would probably not know the chronological order).

I've also noticed that the pull request you have linked for the 0.3.1 entry is for a different feature/bug fix.

Author: @habbes That was a typo. It should be 155 not 115

Author: But then again, moving it to the bottom is better that removing it altogether. I have moved it to the bottom.
This argument won it for me: _someone looking for a specific version would search the changelog in the order of versions, not chronologically (since they would probably not know the chronological order)_

* [ [#115](https://github.com/OData/ODataConnectedService/pull/115) ] Legacy OData Connected Service version that supports previous versions of OData client (versions 6.x). It has been updated to support both VS 2017 and VS 2019

## OData Connected Service 0.10.0

### New Features:
* [ [#83](https://github.com/OData/ODataConnectedService/pull/83) ] Emit dynamic properties container property for open types

### Improvements and Bug Fixes:
* [ [#121](https://github.com/OData/ODataConnectedService/pull/121) ] Fix issue loading generated metadata file when you have multiple files generated in the client project
* [ [#116](https://github.com/OData/ODataConnectedService/pull/116) ] Properly escapes metadata path in the local filesystem when generating T4 templates
* [ [#117](https://github.com/OData/ODataConnectedService/pull/117) ] Fix issue where unselecting an operation import auto-selects schema types that the operation depends on

## OData Connected Service 0.9.1

### New Features:
* [ [#87](https://github.com/OData/ODataConnectedService/pull/87) ] Auto-selects the last metadata endpoint used in generating proxy classes

### Improvements and Bug Fixes:

> * [ [#95](https://github.com/OData/ODataConnectedService/pull/95) ] Enables multiple files generation using design-time T4 templates

Reviewer: ```suggestion
* [ [#95](https://github.com/OData/ODataConnectedService/pull/95) ] Enables generation of multiple files using design-time T4 templates
```
